app/code/Magento/SalesRule/Model/RuleRepository.php

```diff
@@ -5,110 +5,66 @@
-    /**
-     * @var \Magento\Framework\Api\SearchCriteria\CollectionProcessorInterface
-     */
-    private $collectionProcessor;
-
     /**
      * RuleRepository constructor.
      * @param RuleFactory $ruleFactory
-     * @param \Magento\SalesRule\Api\Data\RuleInterfaceFactory $ruleDataFactory
-     * @param \Magento\SalesRule\Api\Data\ConditionInterfaceFactory $conditionDataFactory
-     * @param Converter\ToDataModel $toDataModelConverter
-     * @param Converter\ToModel $toModelConverter
-     * @param \Magento\SalesRule\Api\Data\RuleSearchResultInterfaceFactory $searchResultFactory
-     * @param \Magento\Framework\Api\ExtensionAttribute\JoinProcessorInterface $extensionAttributesJoinProcessor
+     * @param RuleInterfaceFactory $ruleDataFactory
+     * @param ConditionInterfaceFactory $conditionDataFactory
+     * @param ToDataModelConverter $toDataModelConverter
+     * @param ToModelConverter $toModelConverter
+     * @param RuleSearchResultInterfaceFactory $searchResultFactory
+     * @param JoinProcessorInterface $extensionAttributesJoinProcessor
      * @param ResourceModel\Rule\CollectionFactory $ruleCollectionFactory
-     * @param \Magento\Framework\Reflection\DataObjectProcessor $dataObjectProcessor
+     * @param DataObjectProcessor $dataObjectProcessor
      * @param CollectionProcessorInterface|null $collectionProcessor
      * @SuppressWarnings(PHPMD.ExcessiveParameterList)
      */
     public function __construct(
-        \Magento\SalesRule\Model\RuleFactory $ruleFactory,
-        \Magento\SalesRule\Api\Data\RuleInterfaceFactory $ruleDataFactory,
-        \Magento\SalesRule\Api\Data\ConditionInterfaceFactory $conditionDataFactory,
-        \Magento\SalesRule\Model\Converter\ToDataModel $toDataModelConverter,
-        \Magento\SalesRule\Model\Converter\ToModel $toModelConverter,
-        \Magento\SalesRule\Api\Data\RuleSearchResultInterfaceFactory $searchResultFactory,
-        \Magento\Framework\Api\ExtensionAttribute\JoinProcessorInterface $extensionAttributesJoinProcessor,
-        \Magento\SalesRule\Model\ResourceModel\Rule\CollectionFactory $ruleCollectionFactory,
-        \Magento\Framework\Reflection\DataObjectProcessor $dataObjectProcessor,
-        CollectionProcessorInterface $collectionProcessor = null
+        protected readonly RuleFactory $ruleFactory,
+        protected readonly RuleInterfaceFactory $ruleDataFactory,
+        protected readonly ConditionInterfaceFactory $conditionDataFactory,
+        protected readonly ToDataModelConverter $toDataModelConverter,
+        protected readonly ToModelConverter $toModelConverter,
+        protected readonly RuleSearchResultInterfaceFactory $searchResultFactory,
+        protected readonly JoinProcessorInterface $extensionAttributesJoinProcessor,
+        protected readonly RuleCollectionFactory $ruleCollectionFactory,
+        protected readonly DataObjectProcessor $dataObjectProcessor,
+        private ?CollectionProcessorInterface $collectionProcessor = null
```

Comment:
The type of this property seems to have changed by mistake.
```suggestion
        private CollectionProcessorInterface $collectionProcessor = null
```